### ISSUE TITLE
Immediate crash on opening

### ISSUE THREAD

**Issue by MicahN44** [cosmic-utils/web-apps#78]
The title kind of sums it up, but with the flatpak version it crashes immediately no matter how I launch it. The window flashes up, and the icon appears on the dock, and then they both disappear. Does the same no matter how I launch it. 

This could totally be something I have broken to be clear, I am new to this space, but if it is something I have done I am not sure what it is lol

I am using Pop 22.04 with the normal Gnome DE. I may try building it later when I have more time (especially because I should probably know how to do that anyways). thanks

[Screencast from 12-24-2024 07:59:29 PM.webm](https://github.com/user-attachments/assets/a3eb0d2a-fe97-4246-ad8d-aaef9c44ce57)

**Comment by elevenhsoft** [comment]
Hello,

It comes from `winit` directly. There must be something which causes this issue. You are using X11 on this version of PopOS. I had previously some problems with Xorg and libcosmic apps. Mostly because of `winit` but it does not mean it's not fixable. Could you please copy-pase this error message here? I can't see this very well. Also you don't need to build it yourself because the result should be the same. 

Please paste full log here. So I will dig it.

**Comment by MicahN44** [comment]
2024-12-25T14:19:33.239943Z  INFO winit::platform_impl::linux::x11::window: Guessed window scale factor: 1
2024-12-25T14:19:34.293141Z  INFO winit::platform_impl::linux::x11::window: Guessed window scale factor: 1
2024-12-25T14:19:34.305074Z ERROR winit::platform_impl::linux: X11 error: XError {
    description: "BadMatch (invalid parameter attributes)",
    error_code: 8,
    request_code: 149,
    minor_code: 4,
}
2024-12-25T14:19:34.305097Z ERROR winit::platform_impl::linux: X11 error: XError {
    description: "BadMatch (invalid parameter attributes)",
    error_code: 8,
    request_code: 149,
    minor_code: 4,
}
2024-12-25T14:19:34.305107Z ERROR winit::platform_impl::linux: X11 error: XError {
    description: "BadMatch (invalid parameter attributes)",
    error_code: 8,
    request_code: 149,
    minor_code: 4,
}
2024-12-25T14:19:34.363050Z ERROR winit::platform_impl::linux: X11 error: XError {
    description: "BadMatch (invalid parameter attributes)",
    error_code: 8,
    request_code: 149,
    minor_code: 4,
}
2024-12-25T14:19:34.363074Z ERROR winit::platform_impl::linux: X11 error: XError {
    description: "BadMatch (invalid parameter attributes)",
    error_code: 8,
    request_code: 149,
    minor_code: 4,
}
2024-12-25T14:19:34.363082Z ERROR winit::platform_impl::linux: X11 error: XError {
    description: "BadMatch (invalid parameter attributes)",
    error_code: 8,
    request_code: 149,
    minor_code: 4,
}
thread 'main' panicked at /run/build/quick-webapps/cargo/vendor/winit/src/platform_impl/linux/x11/event_processor.rs:1210:47:
Failed to focus input context: XError { description: "BadMatch (invalid parameter attributes)", error_code: 8, request_code: 149, minor_code: 4 }
stack backtrace:
   0:     0x623ae34ef11a - <unknown>
   1:     0x623ae351cdf3 - <unknown>
   2:     0x623ae34ea5c3 - <unknown>
   3:     0x623ae34eef62 - <unknown>
   4:     0x623ae34f004c - <unknown>
   5:     0x623ae34efe92 - <unknown>
   6:     0x623ae34f0627 - <unknown>
   7:     0x623ae34f04ba - <unknown>
   8:     0x623ae34ef5f9 - <unknown>
   9:     0x623ae34f014c - <unknown>
  10:     0x623ae2131200 - <unknown>
  11:     0x623ae2131646 - <unknown>
  12:     0x623ae2316bf0 - <unknown>
  13:     0x623ae23159b6 - <unknown>
  14:     0x623ae2311283 - <unknown>
  15:     0x623ae225944c - <unknown>
  16:     0x623ae225dc1f - <unknown>
  17:     0x623ae2230dd1 - <unknown>
  18:     0x623ae247fd83 - <unknown>
  19:     0x623ae241f9fa - <unknown>
  20:     0x623ae23c89c6 - <unknown>
  21:     0x623ae226f055 - <unknown>
  22:     0x623ae34e255e - <unknown>
  23:     0x623ae241ff25 - <unknown>
  24:     0x77a3d7a2b188 - <unknown>
  25:     0x77a3d7a2b24b - __libc_start_main
  26:     0x623ae2131c15 - <unknown>
  27:                0x0 - <unknown>


Here is everything that was in that terminal, lmk if there is something else I should grab you. And good to know on building it

**Comment by elevenhsoft** [comment]
Thanks!

Are you using `fcitx` maybe?

**Comment by MicahN44** [comment]
That is a good question! I don't know what that is. I don't think so. Checking online it doesn't seem to be built in to Pop

**Comment by elevenhsoft** [comment]
I'm asking cuz found this: https://github.com/rust-windowing/winit/issues/3761

**Comment by MicahN44** [comment]
That certainly looks similar. though AFAIK I don't have fcitx. I will try to verify that

In case it may be helpful here is system info
![Screenshot from 2024-12-25 15-16-14](https://github.com/user-attachments/assets/745ebf88-00ee-456f-a1b1-1930ca8edad4)
Didn't think to include this before.

**Comment by elevenhsoft** [comment]
Thank you!

I will make a VM with this version of PopOS and try to reproduce it on my own.

Also, if it's possible, please try on Wayland. It's simple to enable on PopOS.

**Comment by MicahN44** [comment]
Will give that a Google and try that. I was just looking into it, I had heard that Wayland wasn't friendly with Nvidia but that seems to not be so true anymore? Shows what I know. I will see about doing that real quick :) Thanks for your help btw!

**Comment by elevenhsoft** [comment]
It was not that good before. Now it kind of better. It's basically for test to make sure this issue happens only on X11. Later you can disable it again :)

Thank you again for your testing. Tomorrow I will fire up a VM and do something else :)

**Comment by MicahN44** [comment]
Working on Wayland. I'll try some other stuff and see what happens lol. If worst comes to worst, could I potentially make PWAs in Wayland and then hop back over to X11?

And no rush for my sake :) and no problem. If there is anything I can help with I am more than happy to!

**Comment by elevenhsoft** [comment]
> If worst comes to worst, could I potentially make PWAs in Wayland and then hop back over to X11?

Yes, it should works. But we will manage to make it working on X11 also :)
I noticed you are using cosmic-store which is also baked by libcosmic. So there is something related to Quick Web Apps, something on my end.

**Comment by MicahN44** [comment]
> I noticed you are using cosmic-store which is also baked by libcosmic. So there is something related to Quick Web Apps, something on my end.

Yeah I have been using that for a while because Pop Shop was giving me a ton of trouble. Good catch, I didn't even think about that. Not that that should be surprising to anyone ;)

**Comment by MicahN44** [comment]
I played with Wayland for a while and didn't find any weirdness with other stuff yet, so I will probably stick with that for now, but let me know when you are working on this issue cause I'm happy to do some testing on X11

**Comment by elevenhsoft** [comment]
Hello again @MicahN44 ,

So I did some testing on virtual machine, Pop OS 22.04 and X11. Everything seems to works as expected. However, I did it on default settings.

After some digging, I found from where this error comes: https://github.com/pop-os/winit/blob/iced-0.13/src/platform_impl/linux/x11/event_processor.rs#L1210

It's basically something related to your `input` (?) settings? Could you please tell us your input settings? Also, do you using any of this: https://wiki.archlinux.org/title/Input_method#Input_method_framework
This is related to previous issue I mentioned, about fcitx.

**Comment by MicahN44** [comment]
I don't use any of that, but I may have had KDE Plasma on this install at some point in the past (not sure if I've reinstalled since then). That thread says KDE uses fcitx5 so maybe that is still around somewhere and causing issues? I am at work rn so I can't try anything but I'll poke at it this evening and see what I find

**Comment by elevenhsoft** [comment]
> That thread says KDE uses fcitx5 so maybe that is still around somewhere and causing issues?

I guess so :)

> I am at work rn so I can't try anything but I'll poke at it this evening and see what I find

Sure, no rush! I appreciate your help :)

**Comment by MicahN44** [comment]
OK so I am back on X11 this morning, I am still having issues. It looks like I have IBus from that Archwiki thread (Makes since because I use Gnome) But I think that would have been in your VM as well. I didn't find any commands or running processes for anything else there.

I apologize for my newbiness but what do you mean by my input settings? 

I have a second drive with a Windows install I want to nuke anyways, I was gonna put 24.04 Cosmic on it but I may quickly do a fresh Pop 22.04 install on there and see if I still have issues. Will let you know what happens with that as well

**Comment by MicahN44** [comment]
OK so I am on a fresh Pop install and I am still having this issue. Not sure what would be different between your VM and this. Both of my installs are encrypted but I don't think that should matter?

**Comment by elevenhsoft** [comment]
Hello,

> I apologize for my newbiness but what do you mean by my input settings?

Basically your language, keyboard settings etc. I'm using Polish, and pl keyboard layout.

> OK so I am on a fresh Pop install and I am still having this issue. Not sure what would be different between your VM and this.

Ok, that one is weird. For me it worked. Also, many people using this app and nobody reported anything similar before. I'm not really sure what is going on, but will dig it.

**Comment by MicahN44** [comment]
Keyboard is English US, Language is English US, at least on this second machine. I would assume that's what I did on my actual install as well lol. 

LMK If you want me to try anything else :) idk where to go from here myself. I don't think it would be a hardware issue especially since it works on Wayland.

**Comment by elevenhsoft** [comment]
> Keyboard is English US, Language is English US, at least on this second machine. I would assume that's what I did on my actual install as well lol.

nothing fancy then.

> LMK If you want me to try anything else :) idk where to go from here myself. I don't think it would be a hardware issue especially since it works on Wayland.

Thanks. You did good. I will test it to reproduce and try to fix it. But not sure how if it's works ^^ Yeah, it's X11 issue for sure.

**Comment by elevenhsoft** [comment]
Well, if you are brave enough ^^

You can try to build it yourself. From source. Follow the readme instructions. 

Then you can try with updated dependencies.

**Comment by MicahN44** [comment]
I will give that a shot, that is something I would like to know how to do anyways. Thanks for your help!

**Comment by elevenhsoft** [comment]
Awesome. If you need assistance feel free to ask.

**Comment by MicahN44** [comment]
Sorry to be a bother but I have messed up somewhere along the route lol. the `just` command didn't work properly I don't think (which probably means I did something wrong before that). Here is the error:
![Screenshot from 2024-12-27 12-00-47](https://github.com/user-attachments/assets/0c846146-1c11-4e08-922c-4934b4f116ca)
There was a lot more the first time I ran this, but I restarted my computer just in case that would have helped. this is all the stuff that went wrong the first time. 

From what I can find online this has to do with refutable vs. irrefutable variables? Which is interesting but I wasn't sure where to go from there, or how to use "let else" as the error suggests. 

In addition in the readme it says to build a binary file, I assumed I was doing that but maybe I wasn't? How do I do that and is that my problem?

**Comment by elevenhsoft** [comment]
Ah right. It might be old version of rustc I guess. I was suggesting in README to install rustc with cargo from PopOS repo. It can be problematic. It better for this purpose use rustup to install latest toolchain. Please remove first 2 dependencies as README suggest, so `rustc` and `cargo` and then follow this instruction:

https://rustup.rs/

Later, restart your shell and try to build again :)

Sorry, my bad. I will correct README :)

**Comment by MicahN44** [comment]
That worked perfectly, thank you! And this is working on X11 as well now. If you want to do more testing on the Flatpak let me know but I will stick with this for the moment :)

**Comment by elevenhsoft** [comment]
Oh, well. That's interesting. So there is something on flatpak side. Thanks!

**Comment by hepp3n** [comment]
Hello,

I made some refactoring also updated dependencies. Soon, there will be update to flathub. I hope it will be solved. Sorry for the issue :)

**Comment by MicahN44** [comment]
Sweet! I will test it this evening when I get home if you like. Though full disclosure, I have since reinstalled. It is the same OS but I was having weird issues with suspending and with graphics. But I'll go back to X11 and give it a shot

**Comment by hepp3n** [comment]
Thanks! I'm happy you are still interested ^^ For flathub we need to wait a bit, but master works better now :)

**Comment by MicahN44** [comment]
Oh alright then! Well let me know when you post the flathub version and I can test that too. I'll update and see if the system version works better but I have been having no issues with it so far

**Comment by MicahN44** [comment]
I did not mean to do that lmao
